latex/headers/context.h

```diff
@@ -16,19 +16,33 @@ namespace cl {
 namespace sycl {
 class context {
  public:
-  explicit context(async_handler asyncHandler = {});
+  explicit context(const property_list &propList = {});
 
-  context(const device &dev, async_handler asyncHandler = {});
+  context(async_handler asyncHandler,
+      const property_list &propList = {});
 
-  context(const platform &plt, async_handler asyncHandler = {});
+  context(const device &dev, const property_list &propList = {});
+
+  context(const device &dev, async_handler asyncHandler,
+      const property_list &propList = {});
+
+  context(const platform &plt, const property_list &propList = {});
+
+  context(const platform &plt, async_handler asyncHandler,
+      const property_list &propList = {});
 
   context(const vector_class<device> &deviceList,
-    async_handler asyncHandler = {});
+      const property_list &propList = {});
+
+  context(const vector_class<device> &deviceList,
+      async_handler asyncHandler, const property_list &propList = {});
 
   context(cl_context clContext, async_handler asyncHandler = {});
 
   /* -- common interface members -- */
 
+  /* -- property interface members -- */
```

Comment:
Is it really a property interface here?

Reply:
I'd argue that all objects with property lists should have at least `has_property` and `get_property` to keep it consistent and useful.